File: features/step-definitions/counter-steps.js
import { nthElement } from './utils'

module.exports = function () {
  this.Then(/^the counter value should be "([^"]+)"$/, function (value) {
    return this.browser.assert.text('.counter__value', value)
  })

  this.Then(/^the (\d+)(?:st|nd|rd|th) counter value should be "([^"]+)"$/, function (nth, value) {
    return this.browser.assert.text(
      nthElement(this.browser, nth, 'counter__value'),
      value
    )
  })

  this.Then(/^I should see (\d+) counters?$/, function (count) {
    return this.browser.assert.elements('.counter', count)
  })
}

File: features/step_definitions/counter_steps.rb
Then(/^I should see (\d+) counters?$/) do |n|
  sleep(0.1)
  expect(all('.counter').length).to be(n.to_i)
end

Then(/^the counter value should be "([^"]*)"$/) do |value|
  sleep(0.1)
  expect(page).to have_css('.counter__value', text: value)
end

Then(/^the (\d+)(?:st|nd|rd|th) counter value should be "([^"]*)"$/) do |nth, value|
  sleep(0.1)
  element = nth_element('counter__value', nth)

  expect(element.text).to eq(value)
end
